### PR TITLE
Add timeout parameter to win_service functions

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds a timeout parameter to the following functions in win_service:

- start
- stop
- restart
- delete

Fix some documentation formatting issues

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/48163

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/win_service.py [hunk 1]
```diff
@@ -409,38 +411,33 @@ def start(name):
 
     attempts = 0
     while info(name)['Status'] in ['Start Pending', 'Stopped'] \
-            and attempts <= RETRY_ATTEMPTS:
+            and attempts <= timeout:
```
Comment (damon-atkins) [inline salt/modules/win_service.py]: The time out is doco as seconds not attempts please use the following code 

```
    end_t = time.time() + timeout 
    while info(name)['Status'] in ['Start Pending', 'Stopped'] \
          and time.time() < end_t:
        time.sleep(1)
```

File: salt/modules/win_service.py [hunk 2]
```diff
@@ -450,26 +447,35 @@ def stop(name):
 
     attempts = 0
     while info(name)['Status'] in ['Running', 'Stop Pending'] \
-            and attempts <= RETRY_ATTEMPTS:
+            and attempts <= timeout:
```
Comment (damon-atkins) [inline salt/modules/win_service.py]: dito

File: salt/modules/win_service.py [hunk 3]
```diff
@@ -1282,7 +1319,7 @@ def delete(name):
     win32service.CloseServiceHandle(handle_svc)
 
     attempts = 0
-    while name in get_all() and attempts <= RETRY_ATTEMPTS:
+    while name in get_all() and attempts <= timeout:
```
Comment (damon-atkins) [inline salt/modules/win_service.py]: dito

**Review by damon-atkins — CHANGES_REQUESTED** [review]
Please use  "time code" for timeout rather than attempts

**Comment by damon-atkins** [comment]
open a PR against your fix.

File: salt/modules/win_service.py [hunk 4]
```diff
@@ -100,6 +98,50 @@ def __virtual__():
...
+    '''
+    info_results = info(service_name)
+
+    while info_results['Status'] in service_states and time.time() < end_time:
+        # From Microsoft: Do not wait longer than the wait hint. A good interval
+        # is one-tenth of the wait hint but not less than 1 second and not more
+        # than 10 seconds.
+        # https://docs.microsoft.com/en-us/windows/desktop/services/starting-a-service
+        # https://docs.microsoft.com/en-us/windows/desktop/services/stopping-a-service
+        wait_time = info_results['Status_WaitHint'] / 1000
```
Comment (damon-atkins) [inline salt/modules/win_service.py]: Looks like I got this code slightly wrong.
Should not have the / 1000   see the else which should do this. This way we cover the off chance wait hint is 0.

``
else:
           wait_time = wait_time / 10000  # convert ms to seconds, and then divide by 10.
``

**Comment by twangboy** [comment]
@damon-atkins Could you please review this one last time. I implemented the changes you suggested.

**Review by damon-atkins — APPROVED** [review]
DONE PR update with some more words around the timeout

**Pull request merged**